pilz_robot_programming/src/pilz_robot_programming/commands.py

```diff
@@ -640,9 +640,6 @@ def _to_robot_reference(robot, pose_frame, goal_pose_custom_ref):
 
     robot_ref = robot._robot_commander.get_planning_frame()
 
-    if not _is_quaternion_initialized(goal_pose_custom_ref.orientation):
-        goal_pose_custom_ref.orientation.w = 1
-
```

Comment:
Would it be feasible to delegate this task to moveit? Simply set the `header.frame_id` in your request and leave the transformation to the motion planner. As far as I understand, it should be implemented there (as well) and thus is some code duplication...

Reply:
Sounds good. I will give it a try.

Reply:
> Would it be feasible to delegate this task to moveit?

While in general I would agree with reusing as much as possible, in this case it might not be such a bad thing to try and make sure that things under control of the planner / robot programming framework are consistent.

It's a small thing to make sure that all outgoing planning requests / trajectories / poses are in robot base frame. It also makes the software in this repository more robust against changes in MoveIt (it might stop doing what @jschleicher writes in his comment) and if there is a planning framework that is not MoveIt, but provides the same/a similar interface, but does not transform poses into robot base frame, bad things can happen.

I'm a rather defensive programmer, so this may not be compatible with the style of the rest of `pilz_robot_programming`.

Reply:
@jschleicher @gavanderhoorn Thanks for your inputs. I don't have a strong opinion about it. I would keep it as it is, unless more reviewers request a change.

Reply:
@gavanderhoorn thank you for your input!
I also agree with @jschleicher that the motion planner should be able to handle other frame_ids but maybe we should implement this independent from the need of the underlying PythonAPI.

Reply:
Ok, fine. Keep this for now.